modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -660,11 +660,11 @@ private <T,S> void resizeColumnToFitContent(TableView<T> tv, TableColumn<T, S> t
 
         int rows = maxRows == -1 ? items.size() : Math.min(items.size(), maxRows);
         double maxWidth = 0;
+        cell.updateTableColumn(tc);
+        cell.updateTableView(tv);
         for (int row = 0; row < rows; row++) {
             tableRow.updateIndex(row);
 
-            cell.updateTableColumn(tc);
-            cell.updateTableView(tv);
             cell.updateTableRow(tableRow);
             cell.updateIndex(row);
 
```

Comment:
we could also check if we need/can call `tableRow.applyCss();` just only one time (below) (e.g. at first?),
May works and improve performance, or may not work

Reply:
That doesn't work unfortunately. I tried this:

        boolean tableRowHasCss = false;
        for (int row = 0; row < rows; row++) {
            tableRow.updateIndex(row);
            cell.updateIndex(row);

            if ((cell.getText() != null && !cell.getText().isEmpty()) || cell.getGraphic() != null) {
                if (!tableRowHasCss) {
                    tableRow.applyCss();
                    tableRowHasCss = true;
                }
                maxWidth = Math.max(maxWidth, cell.prefWidth(-1));
            }
        }

but if you have a table with this css:
```
.oddbig .table-row-cell:odd{
    -fx-font-size: 36;
}

```
Then only the css of the first row will be taken into account, i.e. the column wont size correctly.
I've recorded a little video of the effect at https://youtu.be/-p0pv-i4K2s

Reply:
Thanks for checking!